archunit/src/main/java/com/tngtech/archunit/core/domain/JavaClass.java

```diff
@@ -87,14 +89,14 @@
             return result.build();
         }
     });
-    private final Set<JavaClass> interfaces = new HashSet<>();
-    private final Supplier<Set<JavaClass>> allInterfaces = Suppliers.memoize(new Supplier<Set<JavaClass>>() {
+    private Interfaces interfaces = Interfaces.EMPTY;
+    private final Supplier<Set<JavaClass>> allRawInterfaces = Suppliers.memoize(new Supplier<Set<JavaClass>>() {
```

Comment:
Do we additionally need `Supplier<Set<JavaType>> allnterfaces`?

Would we detect that `Child` _is a_ `List<String>` in the following example?
```java
    interface Parent<T> extends List<T> {
    }

    interface Child extends Parent<String> {
    }
```

Reply:
I realized that we don't have the corresponding feature for superclasses either. So this could, of course, be added with another PR, if we want to have it at all (but IMO it would be very useful to handle `JavaTypeVariable`s in the library already).

Reply:
Yeah, I agree that that would be nice, but I'll leave it open for a later time :wink: (and then for generic interfaces as well as superclasses)
The Reflection API doesn't offer that either AFAIK. I see it as nice to have, because all the information to retrieve the actual type arguments of super classes are present. It's just a little tedious right now, since the mapping has to be performed manually (i.e. associate type parameters with actual type arguments, then carry that mapping over to the class hierarchy).
For the Reflection API there are utilities like `org.springframework.core.GenericTypeResolver` though, that do this work for you, while for ArchUnit there is no such easy way, so sooner or later we should probably do it. But it's not essential to get this here out in my view :man_shrugging: